sdcm/provision/scylla_yaml/cluster_builder.py

```diff
@@ -0,0 +1,198 @@
+        if self._is_openldap_authorization:
+            return self._open_ldap_bind_passwd
+        return None
+
+    @property
+    def ldap_url_template(self) -> Optional[str]:
+        if self._is_msldap_authorization:
+            server_port = self._ms_ldap_server_address_port
+            ldap_filter = 'member=CN={USER}'
+        elif self._is_openldap_authorization:
+            server_port = self._open_ldap_server_address_port
+            ldap_filter = 'uniqueMember=uid={USER},ou=Person'
+        else:
+            return None
+        return f'ldap://{server_port}/{self._ldap_base_dn}?cn?sub?({ldap_filter},{self._ldap_base_dn})'
+
+    @property
+    def _is_msldap_authorization(self):
+        return self.params.get('use_ldap_authorization') and self.params.get('use_ms_ad_ldap')
+
+    @property
+    def _is_openldap_authorization(self):
+        return self.params.get('use_ldap_authorization') and not self.params.get('use_ms_ad_ldap')
+
+    @property
+    def _is_ldap_authorization(self):
+        return self.params.get('use_ldap_authorization')
+
+    @property
+    def _ldap_base_dn(self) -> str:
+        return 'dc=scylla-qa,dc=com'
+
+    @property
+    def _ms_ldap_base_dn(self) -> str:
+        return 'DC=scylla-qa,DC=com'
+
+    @builder_property(default_return={'ldap_bind_dn': '', 'admin_password': '', 'server_address': ''})
+    def _msldap_server_info(self) -> dict:
+        self._require_attribute('msldap_server_info', 'MSLDAP is configured, but no `msldap_server_info` is passed')
+        missing_keys = [key for key in ['ldap_bind_dn', 'admin_password', 'server_address'] if key not in
+                        self.msldap_server_info]
+        if missing_keys:
+            raise ValidationError("MSLDAP is configured, but `msldap_server_info` lack of following keys: "
+                                  f"{','.join(missing_keys)}")
+        return self.msldap_server_info
+
+    @property
+    def _ms_ldap_bind_dn(self) -> str:
+        return self._msldap_server_info['ldap_bind_dn']
+
+    @property
+    def _ms_ldap_bind_passwd(self) -> str:
+        return self._msldap_server_info['admin_password']
+
+    @property
+    def _ms_ldap_server_address_port(self) -> str:
+        return f'{self._msldap_server_info["server_address"]}:389'
+
+    @property
+    def _open_ldap_bind_dn(self) -> str:
+        return f'cn=admin,{self._ldap_base_dn}'
+
+    @property
+    def _open_ldap_bind_passwd(self) -> str:
+        return 'scylla-0'
+
+    @builder_property
+    def _open_ldap_server_address_port(self) -> str:
+        if self.params.get('ip_ssh_connections') == 'public' or self._multi_region:
+            # When connection goes public we run ssh tunnel on db-nodes side to access openldap server
+            # that is why we pass address it in scylla config as '127.0.0.1:{LDAP_SSH_TUNNEL_LOCAL_PORT}'
+            from sdcm.utils.ldap import LDAP_SSH_TUNNEL_LOCAL_PORT  # pylint: disable=import-outside-toplevel
```

Comment:
Actually it is bad practice to have imports wrapped anywhere. It postpones import problems if such exist and hides whole import picture.